duckduckgo/iOS
# Duckplayer New Tab + Refactor

Task/Issue URL: https://app.asana.com/0/1204099484721401/1208366641490350/f

**Description**:
- Refactors DuckPlayerNavigation handler for improved navigation and stability
- Implements Open In New Tab Behavior
- Opens "Watch in Youtube" links in new tab by default
- Fixes issues found in Open in New tab closed ship review
- Improves navigation for "Default" mode to prevent full screen videos (https://app.asana.com/0/1204099484721401/1208528683217627/f)
- Updates pixel logic
- Removes DuckPlayer Launch Experiment logic and pixels

ℹ️  Don't be fooled by the number of changes.  Besides the new tab setting, these are primarily small code improvements across the feature and lots of documentation around DuckPlayer itself.

**Steps to test this PR**:

- [ ] Smoke test Duck Player
- [x] Make yourself internal user
- [x] Confirm DuckPlayer "Open in new tab" is present in Duck Player settings
- [ ] Smoke test DuckPlayer.  Confirm YouTube links open in new tab
- [x] Test pixels fire up correctly, as follows:

---

- [x] Launch the app, Set DuckPlayer to 'Always'
- [x] Hit Fire Button
- [x] Search for Metallica videos
- [x]  Confirm duckplayer.view-from.serp is fired

---
- [x] Hit Fire Button
- [x] Go to youtube.com
- [x] Search for metallica
- [x] Tap a video
- [x]  Confirm Pixel fired duckplayer.view-from.youtube.automatic is fired

---
- [x] Hit Fire Button
- [x] Go to https://randomtests.netlify.app/ddg/duckplayer.html
- [x] Tap "Regular Youtube link"
- [x]  Confirm Pixel fired duckplayer.view-from.other is fired



---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1208366641664437
  - https://app.asana.com/0/0/1208544097882662
